packages/api/db/migration/20240808140746_create_animal_identifier_type_table.js

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+const PERMISSION_NAME = 'get:animal_identifier_types';
+
+export const up = async function (knex) {
+  await knex.schema.createTable('animal_identifier_type', (table) => {
+    table.increments('id').primary();
+    table.string('key').notNullable();
+  });
+
+  await knex('animal_identifier_type').insert([
+    { key: 'EAR_TAG' },
+    { key: 'LEG_BAND' },
+    { key: 'OTHER' },
+  ]);
+
+  await knex('permissions').insert({
+    permission_id: 171,
+    name: PERMISSION_NAME,
+    description: 'get animal identifier types',
+  });
+
+  await knex('rolePermissions').insert(
+    [1, 2, 3, 5].map((role_id) => ({ role_id, permission_id: 171 })),
+  );
+};
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+
+export const down = async function (knex) {
+  await knex.schema.dropTable('animal_identifier_type');
+
+  await knex.raw(`
```

Comment:
Maybe something like this is more direct? `await knex('rolePermissions').where('permission_id', 171).del();`

Reply:
I'm more familiar with sql than knex so I have a tendency to just use sql but that looks much better!

Reply:
Yeah true ... from what I can tell the principle behind orms is that by using the portable knex functions we could theoretically switch db providers easily just by changing knex config. Raw functions mayyy need to be updated based upon SQL flavour if we were to switch.

So generally my understanding is it is best to stick with knex where possible and use raw for only limited situations. We haven't been picky though about that at all though!